src/components/ThemeSwitcher/context/SwitchThemeContext.tsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable react-refresh/only-export-components */
+
+import React, {
+  createContext,
+  useState,
+  useEffect,
+  type ReactNode,
+} from "react";
+
+interface ThemeContextProps {
+  theme: string;
+  toggleTheme?: () => void;
+}
+
+export const ThemeContext = createContext<ThemeContextProps>({
+  theme: "light",
+  toggleTheme: () => {},
+});
+
+export const ThemeProvider: React.FC<{ children: ReactNode }> = ({
```

Comment:
Если ты хочешь передать только `children`, то существует специальный тип `PropsWithChildren`
```suggestion
export const ThemeProvider: FC<PropsWithChildren> = ({
```

Также не пиши `React.FC`, а явно импортируй тип `FC`